doc/man3/SSL_shutdown.pod

```diff
@@ -21,161 +21,242 @@ SSL_shutdown, SSL_shutdown_ex - shut down a TLS/SSL or QUIC connection
+exact nature of how a shutdown is performed depends on the underlying protocol
+being used. See the section below pertaining to each protocol for more
+information.
+
+In general, calling SSL_shutdown() in nonblocking mode will initiate the
+shutdown process and return 0 to indicate that the shutdown process has not yet
+completed. Once the shutdown process has completed, subsequent calls to
+SSL_shutdown() will return 1. See the RETURN VALUES section for more
+information.
+
+=head1 TLS AND DTLS-SPECIFIC CONSIDERATIONS
+
+Shutdown for SSL/TLS and DTLS is implemented in terms of the SSL/TLS/DTLS
+close_notify alert message. The shutdown process for SSL/TLS and DTLS
+consists of two steps:
+
+=over 4
+
+=item *
+
+A close_notify shutdown alert message is sent to the peer.
+
+=item *
+
+A close_notify shutdown alert message is received from the peer.
+
+=back
+
+These steps can occur in either order depending on whether the connection
+shutdown process was first initiated by the local application or by the peer.
+
+=head2 Locally-Initiated Shutdown
+
+Calling SSL_shutdown() on a SSL/TLS or DTLS SSL object initiates the shutdown
+process and causes OpenSSL to try to send a close_notify shutdown alert to the
+peer. The shutdown process will then be considered completed once the peer
+responds in turn with a close_notify shutdown alert message.
+
+Calling SSL_shutdown() only closes the write direction of the connection; the
+read direction is closed by the peer. Once SSL_shutdown() is called,
+L<SSL_write(3)> can no longer be used, but L<SSL_read(3)> may still be used
+until the peer decides to close the connection in turn. The peer might
+continuing sending data for some period of time before honouring the local
+application's shutdown request.
+
+=head2 Remotely-Initiated Shutdown
+
+If the peer was the first to initiate the shutdown process by sending a
+close_notify alert message, an application will be notified of this as an EOF
+condition when calling
+L<SSL_read(3)> (i.e., L<SSL_read(3)> will fail and L<SSL_get_error(3)> will
+return B<SSL_ERROR_ZERO_RETURN>), after any application data sent by the peer
+prior to initiating the shutdown has been read. An application should handle
+this condition by calling SSL_shutdown() to respond with a close_notify alert in
+turn, completing the shutdown process, though it may choose to write additional
+application data using L<SSL_write(3)> before doing so. If an application does
+not call SSL_shutdown() in this case, a close_notify alert will not be sent and
+the behaviour will not be fully standards compliant.
+
+=head2 Shutdown Lifecycle
+
+Regardless of whether a shutdown was initiated locally or by the peer, if the
+underlying BIO is blocking, a call to SSL_shutdown() will return either once
+close_notify alert messages have both been sent to and received from the peer,
+and the shutdown process is thereby completed (returning 1), or if an error
+occurs.
```

Comment:
I read that as you only need to call SSL_shutdown() once for a blocking connection. I don't think that's true and you need to call it twice.